components/apimgt/org.wso2.carbon.apimgt.solace/src/main/java/org/wso2/carbon/apimgt/solace/SolaceAdminApis.java

```diff
@@ -141,6 +141,84 @@ public CloseableHttpResponse registerAPI(String organization, String title, Stri
         return null;
     }
 
+    /**
+     * Update API in Solace using AsyncAPI Definition
+     *
+     * @param organization  name of the Organization
+     * @param title         name of the Solace API
+     * @param apiDefinition Async definition of the Solace API
+     * @return CloseableHttpResponse of the PATCH call
+     */
+    public CloseableHttpResponse updateAPIDefinition(String organization, String title, String apiDefinition) {
+        URL serviceEndpointURL = new URL(baseUrl);
+        HttpClient httpClient = APIUtil.getHttpClient(serviceEndpointURL.getPort(), serviceEndpointURL.getProtocol());
+        HttpPatch httpPatch = new HttpPatch(baseUrl + "/" + organization + "/apis/" + title);
+        httpPatch.setHeader(HttpHeaders.AUTHORIZATION, "Basic " + getBase64EncodedCredentials());
+        httpPatch.setHeader(HttpHeaders.CONTENT_TYPE, "text/plain");
+        JsonNode jsonNodeTree;
+        String jsonAsYaml = null;
+        // convert json to yaml
+        try {
+            jsonNodeTree = new ObjectMapper().readTree(apiDefinition);
+            jsonAsYaml = new YAMLMapper().writeValueAsString(jsonNodeTree);
+        } catch (JsonProcessingException e) {
+            log.error(e.getMessage());
```

Comment:
Is it possible to log and proceed if there is parsing errors? Shouldn't we throw an error?

Reply:
If we are not logging the stacktrace, we need to have a proper reason for that too.

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/11001/commits/7a075360aac440e13f3869aa7d808b88b4bf39b4